### PR TITLE
feat(coderd/database): use `template_usage_stats` in `GetUserLatencyInsights` query

### PR DESCRIPTION
**Pull request by mafredri**
This PR updates the `GetUserLatencyInsights` query to use rolled up `template_usage_stats` instead of raw agent and app stats.

### CODE REVIEW

**Comment by mafredri** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/12671?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12671** <a href="https://app.graphite.dev/github/pr/coder/coder/12671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#12670** <a href="https://app.graphite.dev/github/pr/coder/coder/12670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12669** <a href="https://app.graphite.dev/github/pr/coder/coder/12669?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12668** <a href="https://app.graphite.dev/github/pr/coder/coder/12668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12667** <a href="https://app.graphite.dev/github/pr/coder/coder/12667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12666** <a href="https://app.graphite.dev/github/pr/coder/coder/12666?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12665** <a href="https://app.graphite.dev/github/pr/coder/coder/12665?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12664** <a href="https://app.graphite.dev/github/pr/coder/coder/12664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mafredri and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by johnstcn — APPROVED** [review]

File: codersdk/insights.go [hunk 1]
```diff
@@ -98,7 +98,8 @@ type UserActivity struct {
 // ConnectionLatency shows the latency for a connection.
 type ConnectionLatency struct {
 	P50 float64 `json:"p50" example:"31.312"`
-	P95 float64 `json:"p95" example:"119.832"`
```
Comment (mtojek) [inline codersdk/insights.go:101]: Is this change intentional?

Comment (mafredri) [reply codersdk/insights.go:101]: It was, but I reverted it. (It was based on a discussion in Slack where p90/p99 would be preferable over p95, and since this p95 isn't used anywhere, the change would be harmless, but this is not the right place for it.)

**Review by kylecarbs — APPROVED** [review]

**Review by mtojek — APPROVED** [review]

**Comment by mafredri** [comment]
### Merge activity

* **Mar 25, 9:33 AM EDT**: @mafredri started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/coder/coder/12671).
